src/core/pointcloud/qgspointcloudlayer.cpp

```diff
@@ -785,6 +788,23 @@ void QgsPointCloudLayer::calculateStatistics()
     QgsMessageLog::logMessage( QObject::tr( "A statistics calculation task for the point cloud %1 is already in progress" ).arg( this->name() ) );
     return;
   }
+#ifdef HAVE_COPC
+  if ( mDataProvider && mDataProvider->index() && mDataProvider->index()->isValid() && mDataProvider->name() == QStringLiteral( "pdal" ) )
```

Comment:
It seems to me that we do not need to require "pdal" provider here - how about "copc" provider? If someone opens COPC file directly, not through pdal, we should also try to read statistics...